### PR TITLE
refactor(state): simplify connect implementation

### PR DESCRIPTION
**Pull request by mikelgo**
Simplification and elimination of handling the `connect`-overloads.

for reference: https://github.com/rx-angular/rx-angular/pull/1644#discussion_r1408870876

### CODE REVIEW

**Comment by mikelgo** [comment]
On my local machine all tests are passing 🤔

File: libs/state/src/lib/rx-state.service.ts [hunk 1]
```diff
@@ -405,64 +405,104 @@ export class RxState<T extends object> implements OnDestroy, Subscribable<T> {
...
+      projectOrSlices$ &&
+      typeof projectOrSlices$ === 'function' &&
+      !projectValueFn
     ) {
-      this.accumulator.nextSliceObservable(inputOrSlice$);
+      const projectionStateFn = projectOrSlices$;
+      const slice$ = keyOrInputOrSlice$.pipe(
+        map((v) => projectionStateFn(this.accumulator.state, v as V))
+      );
+      this.accumulator.nextSliceObservable(slice$);
```
Comment (edbzn) [inline libs/state/src/lib/rx-state.service.ts:437]: Type error could be fixed like so: 

```suggestion
      this.accumulator.nextSliceObservable(slice$ as Observable<V>);
```

File: libs/state/src/lib/rx-state.service.ts [hunk 2]
```diff
@@ -405,64 +405,104 @@ export class RxState<T extends object> implements OnDestroy, Subscribable<T> {
...
+      isSignal(keyOrInputOrSlice$) &&
+      projectOrSlices$ &&
+      typeof projectOrSlices$ === 'function' &&
+      !projectValueFn
+    ) {
+      const projectionStateFn = projectOrSlices$;
+      const slice$ = toObservable(keyOrInputOrSlice$, {
+        injector: this.injector,
+      }).pipe(map((v) => projectionStateFn(this.accumulator.state, v as V)));
+      this.accumulator.nextSliceObservable(slice$);
```
Comment (edbzn) [inline libs/state/src/lib/rx-state.service.ts:451]: Same here: 

```suggestion
      this.accumulator.nextSliceObservable(slice$ as Observable<V>);
```

**Review by edbzn — COMMENTED** [review]
I love it! :rocket: On my end, I get the errors locally, they could be fixed by following the suggestions. One more thing, I would add a test case to check that the function throws in case of passing wrong params, you could use // @ts-expect-error to disable type checking in the test when passing wrong params. Wdyt?

**Review by edbzn — APPROVED** [review]

**Pull request merged**